Python/flowgraph.c

```diff
@@ -1586,6 +1586,54 @@ optimize_cfg(cfg_builder *g, PyObject *consts, PyObject *const_cache)
     return SUCCESS;
 }
 
+static void
+make_super_instruction(cfg_instr *inst1, cfg_instr *inst2, int super_op)
+{
+    /* Skip if instructions are on different lines */
+    if (inst1->i_loc.lineno != inst2->i_loc.lineno) {
+        return;
+    }
```

Comment:
We can still do this if one of the instructions doesn't have a `lineno`:

```suggestion
    if (0 <= inst1->i_loc.lineno &&
        0 <= inst1->i_loc.lineno &&
        inst1->i_loc.lineno != inst2->i_loc.lineno)
    {
        return;
    }
```